scanners/a11y.py

```diff
@@ -0,0 +1,179 @@
+import boto3
+import json
+import logging
+import os
+
+from scanners import utils
+
+
+workers = 25
+PA11Y_STANDARD = 'WCAG2AA'
+pa11y = os.environ.get("PA11Y_PATH", "pa11y")
+headers = [
+    "redirectedTo",
+    "typeCode",
+    "code",
+    "message",
+    "context",
+    "selector"
+]
+
+
+def get_from_inspect_cache(domain):
+    inspect_cache = utils.cache_path(domain, "inspect")
+    inspect_raw = open(inspect_cache).read()
+    inspect_data = json.loads(inspect_raw)
+    return inspect_data
+
+
+def get_domain_to_scan(inspect_data, domain):
+    domain_to_scan = None
+    redirect = inspect_data.get('redirect', None)
+    if redirect:
+        domain_to_scan = inspect_data.get('redirect_to')
+    else:
+        domain_to_scan = domain
+    return domain_to_scan
+
+
+def get_a11y_cache(domain):
+    return utils.cache_path(domain, "a11y")
+
+
+def domain_is_cached(cache):
+    return os.path.exists(cache)
+
+
+def cache_is_not_forced(options):
+    return options.get("force", False) is False
+
+
+def get_errors_from_pa11y_lambda_scan(domain, cache):
+    client = boto3.client(
+        'lambda',
+        aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID'],
+        aws_secret_access_key=os.environ['AWS_SECRET_ACCESS_KEY'],
+        region_name=os.environ['AWS_REGION_NAME']
```

Comment:
I think these should probably use `os.environ.get()` and then gracefully fail if the values are not set, rather than crash on a Python error about accessing an invalid dict entry.